src/core/display/Container.js

```diff
@@ -428,59 +428,85 @@ Container.prototype.renderWebGL = function (renderer)
         return;
     }
 
-    var i, j;
 
     // do a quick check to see if this element has a mask or a filter.
     if (this._mask || this._filters)
     {
-        renderer.currentRenderer.flush();
-
-        // push filter first as we need to ensure the stencil buffer is correct for any masking
-        if (this._filters && this._filters.length)
-        {
-            renderer.filterManager.pushFilter(this, this._filters);
-        }
+        this.renderAdvancedWebGL(renderer);
+    }
+    else
+    {
+        this._renderWebGL(renderer);
 
-        if (this._mask)
+        // simple render children!
+        for (var i = 0, j = this.children.length; i < j; ++i)
         {
-            renderer.maskManager.pushMask(this, this._mask);
+            this.children[i].renderWebGL(renderer);
         }
+    }
+};
 
-        renderer.currentRenderer.start();
+Container.prototype.renderAdvancedWebGL = function (renderer)
+{
+    renderer.currentRenderer.flush();
 
-        // add this object to the batch, only rendered if it has a texture.
-        this._renderWebGL(renderer);
+    var filters = this._filters;
+    var mask = this._mask;
+    var i, j;
 
-        // now loop through the children and make sure they get rendered
-        for (i = 0, j = this.children.length; i < j; i++)
+    // push filter first as we need to ensure the stencil buffer is correct for any masking
+    if ( filters )
+    {
+        if(!this._enabledFilters)
         {
-            this.children[i].renderWebGL(renderer);
+            this._enabledFilters = [];
         }
 
-        renderer.currentRenderer.flush();
+        this._enabledFilters.length = 0;
 
-        if (this._mask)
+        for (i = 0; i < filters.length; i++)
         {
-            renderer.maskManager.popMask(this, this._mask);
+            if(filters[i].enabled)
+            {
+                this._enabledFilters.push( filters[i] );
```

Comment:
Why not use Array.filter?

 this._enableFilters = filters.filter(function(filter){ return filter.enabled;});